tests/model/test_model.py

```diff
@@ -1019,6 +1019,185 @@ def test_get_latest_message_in_topic(
         assert result == return_value
         self.display_error_if_present.assert_called_once_with(response, self.controller)
 
+    @pytest.mark.parametrize(
+        "user_role, user_type",
+        [
+            (None, "_"),
+            ({"role": 100}, "_owner"),
+            ({"role": 200}, "_admin"),
+            ({"role": 300}, "_moderator"),
+            ({"role": 400}, "_member"),
+            ({"role": 600}, "_guest"),
+        ],
+    )
+    @pytest.mark.parametrize(
+        "realm_editing_settings, expected_response",
+        [
+            case(
+                {
+                    "allow_message_editing": False,
+                    "allow_community_topic_editing": True,
+                },
+                {
+                    "_": ["User not found", False],
+                    "_owner": ["", True],
+                    "_admin": ["", True],
+                    "_moderator": [" Editing messages is disabled", False],
+                    "_member": [" Editing messages is disabled", False],
+                    "_guest": [" Editing messages is disabled", False],
+                },
+                id="editing_msg_disabled:PreZulip4.0",
+            ),
+            case(
+                {
+                    "allow_message_editing": True,
+                    "allow_community_topic_editing": False,
+                },
+                {
+                    "_": ["User not found", False],
+                    "_owner": ["", True],
+                    "_admin": ["", True],
+                    "_moderator": [" Editing topic is disabled", False],
+                    "_member": [" Editing topic is disabled", False],
+                    "_guest": [" Editing topic is disabled", False],
+                },
+                id="editing_topic_disabled:PreZulip4.0",
+            ),
+            case(
+                {
+                    "allow_message_editing": True,
+                    "allow_community_topic_editing": True,
+                },
+                {
+                    "_": ["User not found", False],
+                    "_owner": ["", True],
+                    "_admin": ["", True],
+                    "_moderator": ["", True],
+                    "_member": ["", True],
+                    "_guest": ["", True],
+                },
+                id="editing_topic_disabled:PreZulip4.0",
+            ),
+            case(
+                {"allow_message_editing": True, "edit_topic_policy": 1},
+                {
+                    "_": ["User not found", False],
+                    "_owner": ["", True],
+                    "_admin": ["", True],
+                    "_moderator": ["", True],
+                    "_member": ["", True],
+                    "_guest": [
+                        "Only organization administrators, moderators and members can edit topic",
+                        False,
+                    ],
+                },
+                id="all_users:Zulip4.0+:ZFL75",
```

Comment:
The edit topic policy value and the response appears consistent with the server documentation here (though not the test id?), so based on my other comment regarding TOPIC_EDIT_POLICY not matching documentation, does the code use it correctly?